otherlibs/stdune/digest.ml

```diff
@@ -109,7 +109,15 @@ module Path_digest_result = struct
       Dune_filesystem_stubs.Unix_error.Detailed.equal x y
 end
 
-let path_with_stats ~allow_dirs path (stats : Stats_for_digest.t) :
+exception
+  E of
+    [ `Unix_error of Dune_filesystem_stubs.Unix_error.Detailed.t
+    | `Unexpected_kind
```

Comment:
We can get rid of this ugliness by separating the error constructors from `Path_digest_result.t` into a separate type, and making `Path_digest_result.t = (Digest.t, error) result`.

Reply:
I don't mind introducing `Path_digest_error` and using it in `result` but this is perhaps best done in a separate PR as it would cause a bunch of changes.